AzureFunctions.AngularClient/src/app/functions-list/functions-list.component.ts

```
@@ -11,113 +14,119 @@ import { FunctionInfo } from './../shared/models/function-info';
-                  item.sideNav.cacheService.clearCachePrefix(`https://${scmHostName}`);
-
-                  this._globalStateService.clearBusyState();
-              });
-          }
-      }
+    public viewInfoStream: Subject<TreeViewInfo>;
+    public functions: FunctionNode[] = [];
+    public isLoading: boolean;
+    public functionApp: FunctionApp;
+    public appNode: AppNode;
+
+    private _viewInfoSubscription: RxSubscription;
+
+    private _functionsNode: FunctionsNode;
+
+    constructor(private _globalStateService: GlobalStateService,
+        private _portalService: PortalService,
+        private _translateService: TranslateService,
+        private _broadcastService: BroadcastService
+    ) {
+        this.viewInfoStream = new Subject<TreeViewInfo>();
+
+        this._viewInfoSubscription = this.viewInfoStream
+            .distinctUntilChanged()
+            .switchMap(viewInfo => {
+                this.isLoading = true;
+                this._functionsNode = (<FunctionsNode>viewInfo.node);
+                this.appNode = (<AppNode>viewInfo.node.parent)
+                this.functionApp = this._functionsNode.functionApp;
+                return this._functionsNode.loadChildren();
+            })
+            .subscribe(() => {
+                this.isLoading = false;
+                this.functions = (<FunctionNode[]>this._functionsNode.children);
+            });
+    }
+
+    ngOnInit() {
+    }
+
+    ngOnDestroy(): void {
+        this._viewInfoSubscription.unsubscribe();
+    }
+
+    @Input() set viewInfoInput(viewInfo: TreeViewInfo) {
+        this.viewInfoStream.next(viewInfo);
+    }
+
+    clickRow(item: FunctionNode) {
+        item.select();
+    }
+
+    enableChange(item: FunctionNode) {
+        item.functionInfo.config.disabled = !item.functionInfo.config.disabled;
+        return this.functionApp.updateFunction(item.functionInfo)
+            .do(null, e => {
```

It would also be nice if you call clear the error that's posted below on success. That way if the user attempts to click on it again, and this time it works, the error will be hidden. To do that set this as the first do parameter

```ts
_ => this._broadcastService.broadcast<string>(BroadcastEvent.ClearError, ErrorIds.failedToSwitchEnabledFunction)
```